lib/feature/home/widget/quran_page_widget.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:quran/common/bloc/page_bloc.dart';
+
+class QuranPageWidget extends StatefulWidget {
+  @override
+  _QuranPageWidgetState createState() => _QuranPageWidgetState();
+}
+
+class _QuranPageWidgetState extends State<QuranPageWidget> {
+  PageBloc _pageBloc;
+  @override
+  void dispose() {
+    _pageBloc.close();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _pageBloc = BlocProvider.of<PageBloc>(context);
+    return BlocBuilder<PageBloc, int>(
+      builder: (context, current) {
+        print(current);
+        return PageView.builder(
+          controller: PageController(initialPage: current, keepPage: false),
+          itemCount: 605,

```

Comment:
here itemCount should come from a constant class, do not hardcode this way, and what do you think about loading 605 pages at a single time? don't you think it will take a lots of memory.

Reply:
I'd use SwipeDetector instead.